core/lib/src/cookies.rs

```diff
@@ -259,15 +259,15 @@ impl<'a> CookieJar<'a> {
     /// ```
     pub fn get_pending(&self, name: &str) -> Option<Cookie<'static>> {
         let ops = self.ops.lock();
-        for op in ops.iter().rev().filter(|op| op.cookie().name() == name) {
```

Comment:
We need to return as soon as possible from here so we drop the lock. This implementation makes that fact less obvious. Please revert.